droidlet/perception/robot/handlers/label_propagate.py

```diff
@@ -2,21 +2,57 @@
 Copyright (c) Facebook, Inc. and its affiliates.
 """
 from .core import AbstractHandler
-from tokenize import String
 import numpy as np
 import os
-import sys
-if "/opt/ros/kinetic/lib/python2.7/dist-packages" in sys.path:
-    sys.path.remove("/opt/ros/kinetic/lib/python2.7/dist-packages")
 import cv2
 import json
-from copy import deepcopy as copy
-from IPython import embed
-import time
-from scipy.spatial.transform import Rotation
 import glob
-import argparse
 from droidlet.lowlevel.robot_mover_utils import transform_pose
+from numba import njit
+from math import ceil, floor
+
+def compute_uvone(height, width):
+    intrinsic_mat = np.array([[256, 0, 256], [0, 256, 256], [0, 0, 1]])
+    rot = np.array([[0.0, 0.0, 1.0], [-1.0, 0.0, 0.0], [0.0, -1.0, 0.0]])
+    trans = np.array([0, 0, 0.6])
+    intrinsic_mat_inv = np.linalg.inv(intrinsic_mat)
+    img_resolution = (height, width)
+    img_pixs = np.mgrid[0 : img_resolution[0] : 1, 0 : img_resolution[1] : 1]
+    img_pixs = img_pixs.reshape(2, -1)
+    img_pixs[[0, 1], :] = img_pixs[[1, 0], :]
+    uv_one = np.concatenate((img_pixs, np.ones((1, img_pixs.shape[1]))))
+    uv_one_in_cam = np.dot(intrinsic_mat_inv, uv_one)
+    return uv_one_in_cam, intrinsic_mat, rot, trans
+
+def convert_depth_to_pcd(depth, pose, uv_one_in_cam, rot, trans):
+    # point cloud in camera frame
+    depth = (depth.astype(np.float32) / 1000.0).reshape(-1)
+    pts_in_cam = np.multiply(uv_one_in_cam, depth)
+    pts_in_cam = np.concatenate((pts_in_cam, np.ones((1, pts_in_cam.shape[1]))), axis=0)
+    # point cloud in robot base frame
+    pts_in_base = pts_in_cam[:3, :].T
+    pts_in_base = np.dot(pts_in_base, rot.T)
+    pts_in_base = pts_in_base + trans.reshape(-1)
+    # point cloud in world frame (pyrobot)
+    pts_in_world = transform_pose(pts_in_base, pose)
+    return pts_in_world
+
+@njit
+def get_annot(height, width, pts_in_cur_img, src_label):
+    annot_img = np.zeros((height, width))
+    for x in range(len(pts_in_cur_img)):
+        r = int(x/width)
+        c = int(x - r*width)
+        x, y, _ = pts_in_cur_img[x]
```

Comment:
is it necessary to use x as loop var and as coordinate?

Reply:
No, good catch.